src/common/observablevector.ts

```diff
@@ -322,6 +330,11 @@ class ObservableVector<T> extends Vector<T> implements IObservableVector<T> {
    */
   set(index: number, value: T): void {
     let oldValues = [this.at(index)];
+    // Bail if the value does not change.
+    let itemCmp = this._itemCmp;
+    if (itemCmp(oldValues[0], value)) {
```

Comment:
alternatively, you could actually *do* the swap in `set`, but only emit if there is a change. That would make my example work, but I could maybe craft another that wouldn't